From b6dc3c5ae65d69ec97e619ef739bd0221ed1efb2 Mon Sep 17 00:00:00 2001
From: Andrew Purtell <apurtell@apache.org>
Date: Fri, 15 Jan 2016 18:42:29 -0800
Subject: [PATCH 031/197] Amend HBASE-14512 Cache UGI groups

Handle the case where tests might create a call with a null connection
---
 .../org/apache/hadoop/hbase/ipc/RpcServer.java     |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
index 4797143..27818fe 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
@@ -338,7 +338,7 @@ public class RpcServer implements RpcServerInterface, ConfigurationObserver {
       this.isError = false;
       this.size = size;
       this.tinfo = tinfo;
-      this.user = connection.user;
+      this.user = connection == null ? null : connection.user;
       this.remoteAddress = remoteAddress;
       this.retryImmediatelySupported = connection.retryImmediatelySupported;
     }
-- 
1.7.0.4

